docs/rfcs/experimental-profiling.md

```diff
@@ -0,0 +1,86 @@
+# Experimental support for profiling
+
+## Overview
+
+The OpenTelemetry collector has traces, metrics and logs as stable signals. We
+want to start experimenting with support for profiles as an experimental
+signal. But we also don't want to introduce breaking changes in packages
+otherwise considered stable.
+
+This document describes:
+
+* The approach we intend to take to introduce profiling with no breaking changes
+* How the migration will happen once profiling goes stable
+
+### Discarded approach
+
+A first approach, discussed in [issue
+10207](https://github.com/open-telemetry/opentelemetry-collector/issues/10207)
+has been discarded.
+It aimed to refactor the current packages with per-signal subpackages, so each
+subpackage could have its own stability level, like pdata does.
+
+This has been discarded, as the Collector SIG does not want the profiling
+signal to impact the road to the collector reaching 1.0.
+
+## Proposed approach
+
+The proposed approach will consist of two main phases:
+
+* Introduce `experimental` packages for each required module of the collector that needs to be profiles-aware.
+	* `consumer`, `receiver`, `connector`, `component`, `processor`
+* Mark specific APIs as `experimental` in their godoc for parts that can't be a new package.
+	* `service`
+
+### Introduce `experimental` packages
+
+Each package that needs to be profiling signal-aware will have its barebones copied into
+a new `XXexperimental` package (eg: `consumerexperimental`) which will be able
+to handle the profiling signal.
+
+The stable related stable packages MUST not be aware of their experimental
+counterpart.
+The experimental package MAY call its stable implementation.
+
+### Mark specific APIs as `experimental`
+
+In order to boot a functional collector with profiles support, some stable
```

Comment:
Have you explored any of the alternatives listed on https://github.com/open-telemetry/opentelemetry-collector/issues/10207#issuecomment-2135051693 ? Do you have an idea of what it would entail to make service signal-agnostic?

Not saying this is not the right solution, just that I would like to see a more thorough justification of why the alternatives are not preferred

Reply:
I've been trying not to make any changes to stable packages, to go with the requirement to be able to move as fast as possible to v1.
So the proposals `Having two Service constructors` and `Have an unstable Setting for enabling profiles` seems hard to do, since they would require some changes to the public API.

As for `Have all functionality related to profiles under a build tag`, the problem with build tags is that they can't be specified for library imports.
So that would make it very hard to start building receivers and connectors.

Reply:
`Having two Service constructors` would not require changes to the public API, only to the internals. 

I can't speak for other maintainers (I guess to get a clear answer we would need to have a clear picture of the changes in question), but I personally am okay with public API changes to Service if they are a clear win, since `service` as a Go module is still far from being marked as 1.0.

Agreed with the rest (build tags lead to a bad UX for that very reason), but I think we should include those details in the text of the proposal itself.

Reply:
I have changed the RFC to follow https://github.com/open-telemetry/opentelemetry-collector/pull/10253 instead.